### ISSUE TITLE
[Bug] [a11y] Consider increasing the "Scan" and "Search engine" options

### ISSUE THREAD

**Issue by lobontiumira** [mozilla-mobile/fenix#16829]
## Steps to reproduce
1. Have the Accessibility scanner app installed and running.
2. Tap on "+" in order to open a new tab, then scan the page containing the "Scan" and "Search engine" options.

### Expected behavior
No suggestions are made.

### Actual behavior
The Accessibility scanner suggests to increase the "Scan", and "Search engine" items from 40 dp, to 48 dp or larger.

### Device information
* Android device: Samsung Galaxy Tab S6 (Android 10)
* Fenix version: Beta 84.0.0-beta.3, and 12/2 Nightly
 
![Untitled](https://user-images.githubusercontent.com/40694158/100984574-913a3200-3553-11eb-840e-24a9d23f46dd.png)

**Comment by Debanshu777** [comment]
@Amejia481 I would like to work on this issue

Is it suggested to just increase the size of the View from 40dp to 48dp?
 
<img src="https://user-images.githubusercontent.com/50791485/102749436-07c79400-438a-11eb-80ef-067aed2d8799.jpg" height=300 width= 500/>

**Comment by Amejia481** [comment]
> @Amejia481 I would like to work on this issue
> 
> Is it suggested to just increase the size of the View from 40dp to 48dp?
>  
> <img src="https://user-images.githubusercontent.com/50791485/102749436-07c79400-438a-11eb-80ef-067aed2d8799.jpg" height=300 width= 500/>
> 
>  Hi @Debanshu777 👋,
I'm currently on PTO, feel free to work on the issue, others team members will be there to assist you in case you need any help :)

**Comment by sflorean** [comment]
This is still reproducible on all builds, device: Huawei P40 Lite (Android 11).

**Comment by eliserichards** [comment]
I think we should double check with UX that this is ok with them. Tagging for feedback...

**Comment by Dunexus** [comment]
@eliserichards
I don't know much about current priorities but this may not be relevant if https://github.com/mozilla-mobile/fenix/issues/16659 and https://github.com/mozilla-mobile/fenix/issues/16658 are planned to be implemented soon.

**Comment by topotropic** [comment]
Is it possible to increase the touch area while keeping the design as is? 

I'll let @vesta0 comment on when we plan to work on below changes.

> I don't know much about current priorities but this may not be relevant if #16659 and #16658 are planned to be implemented soon.

**Comment by vesta0** [comment]
Going back to @topotropic 's question: Is it possible to increase the touch area while keeping the design as is?

**Comment by Meenohara** [comment]
Please assign this to me

**Comment by Meenohara** [comment]
I have the fenix code cloned from github on April 12 2021 and on searching fragment_search_dialog.xml,  the layout_height still shows  40dp.  
So I visited [17162](https://github.com/mozilla-mobile/fenix/pull/17162) to check if the fix has been applied or not.

1. The merging has been blocked
2. [Issue](https://github.com/mozilla-mobile/fenix/pull/17162#issuecomment-759020194)  is not being discussed
3. Need clarification on this [proposal](https://github.com/mozilla-mobile/fenix/pull/17162#issuecomment-802331634). Is this a part of the bug now?

**Comment by eliserichards** [comment]
> 3. Need clarification on this proposal. Is this a part of the bug now?

Yes, I think @topotropic's [comment](https://github.com/mozilla-mobile/fenix/pull/17162#issuecomment-802331634) on that issue is the same question that she posted above. Can we make the clickable area bigger without changing the visual design of the button? That should be investigated to see if it is possible

**Comment by Meenohara** [comment]
What about https://github.com/mozilla-mobile/fenix/pull/17162#issuecomment-759020194 mentioning all instances of a button?

**Comment by eliserichards** [comment]
Let's see if it's possible on this button first. If it is possible, then we can spread it to the other buttons :) I think there is an issue open for making all of our buttons consistent.. let me find that and link it here

**Comment by Meenohara** [comment]
@eliserichards #17162 has been blocked from merging, needed clarification that too

**Comment by Meenohara** [comment]
1. In the current codebase the dimensions are still 40dp, so do we leave it unchanged?

2. Can we use TouchDelegate API? I have not attempted this on the code yet during the time of writing this comment. 
 This would involve doing it programmatically, than changing dimensions in the xml file
 
 What I have already tried: Playing with the padding. Not helpful, was finding it difficult to test and consumed a lot of time as my Android studio kept slowing down way too much. And I could not ascertain if it was able to override the style or not. Researching this, and need some time to do again on my machine,

 Which approach should we go with? Touch Delegate or Padding /Margin?

**Comment by Meenohara** [comment]
@eliserichards are the above mentioned points ok?
Is there some other way of doing this I haven't thought of?

**Comment by eliserichards** [comment]
Implementing the [TouchDelegate API](https://developer.android.com/reference/android/view/TouchDelegate) would be a great way to handle this. I think that would greatly enhance our accessibility, and would help us with [this work](https://github.com/mozilla-mobile/fenix/issues/19196) as well.

**Comment by Meenohara** [comment]
If we use TouchDelegate API, we would have to make chages in SearchDialogFragment.kt and perhaps other places too

but how would that impact [19196](https://github.com/mozilla-mobile/fenix/issues/19196)

**Comment by Meenohara** [comment]
I have added this code for qr_scan_button  in SearchDialogFragment.kt :

val parent = qr_scan_button.getParent() as View//use pill_wrapper instead?
            parent.post {
                val hitRect = Rect()
                parent.getHitRect(hitRect)
                hitRect.right = hitRect.right - hitRect.left
                hitRect.bottom = hitRect.bottom - hitRect.top
                hitRect.top = 15
                hitRect.left = 15
                logDebug("Troubleshooting", "In TouchDelegate API")
                parent.touchDelegate = TouchDelegate(hitRect, qr_scan_button)
            }


And for Search engine button in fragment_search_dialog.xml:

   <ToggleButton
        android:id="@+id/search_engines_shortcut_button"
        style="@style/search_pill"
        android:textOff="@string/search_engine_button"
        android:textOn="@string/search_engine_button"
        app:drawableStartCompat="@drawable/ic_search"
        android:paddingLeft="10dp"
        android:paddingRight="10dp"
        android:paddingBottom="10dp"
        android:paddingTop="10dp"
        android:layout_marginLeft="-10dp"
        android:layout_marginRight="-10dp"
        android:layout_marginBottom="-10dp"
        android:layout_marginTop="-10dp"
        app:layout_constraintBottom_toBottomOf="@id/pill_wrapper"
        app:layout_constraintStart_toEndOf="@id/qr_scan_button"
        app:layout_constraintTop_toTopOf="@id/pill_wrapper" />

**Comment by eliserichards** [comment]
> Is 19196 not about button components other than only Search and Scan?

Yes, you are correct. #19196 is about buttons in general, not these specific buttons.

**Comment by eliserichards** [comment]
This is available if someone wants to work on it 😄 @Meenohara has done the initial research into using the [TouchDelegate API ](https://developer.android.com/reference/android/view/TouchDelegate), and we would like to continue with this approach.

**Comment by kaream671** [comment]
16826

**Comment by neha-b2001** [comment]
Hello @eliserichards, can I please work on this issue? I have read all the comments associated with the issue, and will proceed using TouchDelegate API.

**Comment by lobontiumira** [comment]
This is still reproducible on 11/2 Nightly build with Google Pixel (Android 10).

**Comment by gabrielluong** [comment]
Closing as a duplicate of https://github.com/mozilla-mobile/fenix/issues/24461
